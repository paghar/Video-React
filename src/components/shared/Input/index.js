//@flow
import React from 'react';
import { StyledInput, StyledSpan, StyledDiv } from './index.style';
type Font = {
  fontSize: Object | string,
  fontName: Object | string,
  fontWeight: Object | string,
  lineHeight: Object | string,
};
type Props = {
  font: Font,
  rtl?: boolean,
  changed?: (event: SyntheticInputEvent<HTMLInputElement>) => void,
  focused?: (event: SyntheticInputEvent<HTMLInputElement>) => void,
  iconClicked?: (event: SyntheticInputEvent<HTMLInputElement>) => void,
  blured?: (event: SyntheticInputEvent<HTMLInputElement>) => void,
  width?: string,
  height?: string,
  borderRadius: string,
  textColor: string,
  backgroundColor: string,
  hoverBackgroundColor?: string,
  placeHolderColor?: string,
  placeholder: string,
  icon?: string,
  iconColor?: string,
  iconSize?: string,
  paddingRL: string,
  className?: string,
  value?: string,
};
const Input = ({
  font: { fontName, fontSize, fontWeight, lineHeight },
  rtl,
  width,
  height,
  changed,
  focused,
  iconClicked,
  blured,
  borderRadius,
  textColor,
  backgroundColor,
  hoverBackgroundColor,
  placeHolderColor,
  placeholder,
  icon,
  iconColor,
  iconSize,
  paddingRL,
  className,
  value,
}: Props) => {
  return (
    <StyledDiv
      rtl={rtl}
      height={height}
      width={width}
      borderRadius={borderRadius}
      backgroundColor={backgroundColor}
      hoverBackgroundColor={hoverBackgroundColor}
      fontName={fontName}
      sizing={fontSize}
      paddingRL={paddingRL}
      weight={fontWeight}
      lineHeight={lineHeight}
      className={className}
    >
      <StyledInput
        textColor={textColor}
        rtl={rtl}
        onChange={changed}
        onFocus={focused}
        onBlur={blured}
        placeHolderColor={placeHolderColor}
        placeholder={placeholder}
        iconSize={iconSize}
        value={value}
      />
      <StyledSpan
        paddingRL={paddingRL}
        rtl={rtl}
        fontSize={iconSize}
        color={iconColor}
        clicked={iconClicked}
        iconNumber={icon}
      />
    </StyledDiv>
  );
};
export default Input;
