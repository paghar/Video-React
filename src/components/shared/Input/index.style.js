//@flow
import styled from 'styled-components';
import Icon from 'components/shared/Icon';
export const StyledInput = styled.input`
  outline: none;
  border: none;
  background: none;
  width: 100%;
  color: ${({ textColor }) => textColor};
  flex-grow: 1;
  ::placeholder {
    color: ${({ placeHolderColor }) => placeHolderColor};
  }
  :-ms-input-placeholder {
    color: ${({ placeHolderColor }) => placeHolderColor};
  }
  ::-moz-input-placeholder {
    color: ${({ placeHolderColor }) => placeHolderColor};
  }
  :hover {
    background-color: ${({ hoverBackgroundColor }) => hoverBackgroundColor};
  }
`;
export const StyledSpan = styled(Icon)`
  padding-right: ${({ rtl, paddingRL }) => (rtl ? paddingRL : '0px')};
  padding-left: ${({ rtl, paddingRL }) => (rtl ? '0px' : paddingRL)};
  color: ${({ color }) => color};
  font-size: ${({ fontSize }) => fontSize};
  align-self: center;
`;
export const StyledDiv = styled.div`
  box-sizing: border-box;
  align-content: stretch;
  direction: ${({ rtl }) => (rtl ? 'rtl' : 'ltr')};
  border-radius: ${({ borderRadius }) => borderRadius};
  background-color: ${({ backgroundColor }) => backgroundColor};
  font-family: ${({ fontName }) => fontName};
  font-size: ${({ sizing }) => sizing};
  font-weight: ${({ weight }) => weight};
  line-height: ${({ lineHeight }) => lineHeight};
  width: ${({ width }) => width};
  height: ${({ height }) => height};
  padding-right: ${({ paddingRL }) => paddingRL};
  padding-left: ${({ paddingRL }) => paddingRL};
  display: flex;
`;
