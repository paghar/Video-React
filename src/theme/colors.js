import theme from 'styled-theming';

export const Generals = {
  i: theme('mode', {
    dark: '#05050d',
  }),
  ii: theme('mode', {
    dark: '#0d0d1a',
  }),
  iii: theme('mode', {
    dark: '#151526',
  }),
  iv: theme('mode', {
    dark: '#212133',
  }),
  v: theme('mode', {
    dark: '#262640',
  }),
  vi: theme('mode', {
    dark: '#7b7b9a',
  }),
  vii: theme('mode', {
    dark: '#b7b7cc',
  }),
  viii: theme('mode', {
    dark: '#ffffff',
  }),
  x: theme('mode', {
    dark: '#e6e6f3',
  }),
  ix: theme('mode', {
    dark: '#f1f1f9',
  }),
};

export const Primary = {
  Dim: theme('mode', {
    dark: '#2172ff',
  }),
  Light: theme('mode', {
    dark: '#317BFC',
  }),
  Dark: theme('mode', {
    dark: '#0E5EE8',
  }),
};

export const PromptWarn = {
  Dim: theme('mode', {
    dark: '#F51653',
  }),
  Light: theme('mode', {
    dark: '#FF4074',
  }),
  Dark: theme('mode', {
    dark: '#DE0540',
  }),
};

export const Secondary = {
  i: theme('mode', {
    dark: '#00d3b2',
  }),
  ii: theme('mode', {
    dark: '#ffcc00',
  }),
  iii: theme('mode', {
    dark: '#f51653',
  }),
  iv: theme('mode', {
    dark: '#64a7ff',
  }),
};

export const Avatars = [
  theme('mode', {
    dark: '#5c4dff',
  }),
  theme('mode', {
    dark: '#685ce6',
  }),
  theme('mode', {
    dark: '#7d73e6',
  }),
  theme('mode', {
    dark: '#928ae6',
  }),
];

export const Gradient = theme('mode', {
  dark:
    'linear-gradient(to bottom, rgba(21, 21, 38, 0), rgba(21, 21, 38, 0.69) 53%, #111120)',
});
