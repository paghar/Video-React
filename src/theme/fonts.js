import theme from 'styled-theming';

export const Title1 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '22px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '35px',
  }),
};

export const Title2 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '21px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '33px',
  }),
};

export const Display1 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '14px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '23px',
  }),
};

export const Display1M = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '14px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '23px',
  }),
};

export const Display2 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '13px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '21px',
  }),
};
export const Display3 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '13px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '21px',
  }),
};
export const Display4 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '12px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '20px',
  }),
};
export const Display5 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '11px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '18px',
  }),
};
export const Display6 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '11px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '18px',
  }),
};
