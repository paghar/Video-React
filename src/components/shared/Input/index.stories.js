//@flow
import React from 'react';
import { action } from '@storybook/addon-actions';
import Input from '.';
import { Display3 } from 'theme/fonts';

export default { title: 'Input' };
export const InputSearch = () => (
  <Input
    height="31px"
    width="223px"
    font={Display3}
    borderRadius="40px"
    textColor="#C4C3D9"
    backgroundColor="#212133"
    hoverBackgroundColor="#000"
    placeholder="Search here..."
    placeHolderColor="#7b7b9a"
    icon="61442"
    iconColor="#7b7b9a"
    iconSize="15px"
    paddingRL="14px"
    changed={action('changed')}
    focused={action('focused')}
    blured={action('blured')}
  />
);
export const ReverseInput = () => (
  <Input
    height="31px"
    width="223px"
    font={Display3}
    rtl
    paddingRL="14px"
    borderRadius="40px"
    iconSize="15px"
    textColor="#C4C3D9"
    backgroundColor="#212133"
    hoverBackgroundColor="#000"
    placeHolderColor="#7b7b9a"
    placeholder="Results"
    icon="61703"
    iconColor="#7b7b9a"
    changed={action('changed')}
    focused={action('focused')}
    blured={action('blured')}
  />
);
