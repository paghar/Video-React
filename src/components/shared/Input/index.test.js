//@flow
import { shallow, mount } from 'enzyme';
import React from 'react';
import Input from '.';
import renderer from 'react-test-renderer';
import { Display3 } from 'theme/fonts';

describe('<Input />', () => {
  it('checks the UI', () => {
    const wrapper = renderer
      .create(
        <Input
          height="53px"
          width="383px"
          font={Display3}
          rtl
          paddingRL="25px"
          borderRadius="40px"
          iconSize="22px"
          textColor="#7b7b9a"
          backgroundColor="#212133"
          hoverBackgroundColor="#000"
          placeholder="Results"
          icon="61703"
          iconColor="#7b7b9a"
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('checks the reverse UI', () => {
    const wrapper = renderer
      .create(
        <Input
          height="53px"
          width="383px"
          font={Display3}
          borderRadius="40px"
          textColor="#7b7b9a"
          backgroundColor="#212133"
          hoverBackgroundColor="#000"
          placeholder="Search here..."
          icon="61442"
          iconColor="#7b7b9a"
          iconSize="22px"
          paddingRL="25px"
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('checks change event', () => {
    const mockCallBack = jest.fn();
    const wrapper = mount(
      <Input
        height="53px"
        width="383px"
        font={Display3}
        borderRadius="40px"
        textColor="#7b7b9a"
        backgroundColor="#212133"
        hoverBackgroundColor="#000"
        placeholder="Search here..."
        icon="61442"
        iconColor="#7b7b9a"
        iconSize="22px"
        paddingRL="25px"
        changed={mockCallBack}
      />
    );
    wrapper.find('input').simulate('change');
    expect(mockCallBack).toHaveBeenCalled();
  });
  it('checks blur event', () => {
    const mockCallBack = jest.fn();
    const wrapper = mount(
      <Input
        height="53px"
        width="383px"
        font={Display3}
        borderRadius="40px"
        textColor="#7b7b9a"
        backgroundColor="#212133"
        hoverBackgroundColor="#000"
        placeholder="Search here..."
        icon="61442"
        iconColor="#7b7b9a"
        iconSize="22px"
        paddingRL="25px"
        blured={mockCallBack}
      />
    );
    wrapper.find('input').simulate('blur');
    expect(mockCallBack).toHaveBeenCalled();
  });
  it('checks focus event', () => {
    const mockCallBack = jest.fn();
    const wrapper = mount(
      <Input
        height="53px"
        width="383px"
        font={Display3}
        borderRadius="40px"
        textColor="#7b7b9a"
        backgroundColor="#212133"
        hoverBackgroundColor="#000"
        placeholder="Search here..."
        icon="61442"
        iconColor="#7b7b9a"
        iconSize="22px"
        paddingRL="25px"
        focused={mockCallBack}
      />
    );
    wrapper.find('input').simulate('focus');
    expect(mockCallBack).toHaveBeenCalled();
  });
});
