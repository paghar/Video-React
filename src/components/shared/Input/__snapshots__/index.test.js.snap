// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> checks the UI 1`] = `
.c2 {
  font-family: 'FontAwesome' !important;
  font-size: 22px;
  color: #7b7b9a;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  color: #7b7b9a;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  color: #7b7b9a;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  color: #7b7b9a;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  color: #7b7b9a;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  padding-right: 25px;
  padding-left: 0px;
  color: #7b7b9a;
  font-size: 22px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 {
  box-sizing: border-box;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  direction: rtl;
  border-radius: 40px;
  background-color: #212133;
  width: 383px;
  height: 53px;
  padding-right: 25px;
  padding-left: 25px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
  height="53px"
  width="383px"
>
  <input
    className="c1"
    placeholder="Results"
  />
  <span
    className="c2 c3"
    fontSize="22px"
  >
    
  </span>
</div>
`;

exports[`<Input /> checks the reverse UI 1`] = `
.c2 {
  font-family: 'FontAwesome' !important;
  font-size: 22px;
  color: #7b7b9a;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  color: #7b7b9a;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  color: #7b7b9a;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  color: #7b7b9a;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  outline: none;
  border: none;
  background: none;
  width: 100%;
  color: #7b7b9a;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  padding-right: 0px;
  padding-left: 25px;
  color: #7b7b9a;
  font-size: 22px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 {
  box-sizing: border-box;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  direction: ltr;
  border-radius: 40px;
  background-color: #212133;
  width: 383px;
  height: 53px;
  padding-right: 25px;
  padding-left: 25px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
  height="53px"
  width="383px"
>
  <input
    className="c1"
    placeholder="Search here..."
  />
  <span
    className="c2 c3"
    fontSize="22px"
  >
    
  </span>
</div>
`;
